src/sardana/macroserver/test/res/macros/testmacros.py

```diff
@@ -0,0 +1,148 @@
+from sardana.macroserver.macro import Type, Macro
+
+MSG_TEMP = "Parsing failed (result: %r; expected: %r)"
+
+
+class runMacro(Macro):
+
+    def run(self, *args):
+
+        expect_params = (99, [1., 2.])
+        macro, _ = self.prepareMacro("pt6_base", *expect_params)
+        self.runMacro(macro)
+        result = macro.data
+        msg = MSG_TEMP % (result, expect_params)
+        assert expect_params == result, msg
+
+        params = (99, 1., 2.)
+        macro, _ = self.prepareMacro("pt6_base", *params)
+        expect_params = (99, [1., 2.])
+        self.runMacro(macro)
+        result = macro.data
+        msg = MSG_TEMP % (result, expect_params)
+        assert expect_params == result, msg
+
+        expect_params = ([92], True)
+        macro, _ = self.prepareMacro("pt10_base", *expect_params)
+        self.runMacro(macro)
+        result = macro.data
+        msg = MSG_TEMP % (result, expect_params)
+        assert expect_params == result, msg
+
+        params = (91, True)
+        expect_params = ([91], True)
+        macro, _ = self.prepareMacro("pt10_base", *params)
+        self.runMacro(macro)
+        result = macro.data
+        msg = MSG_TEMP % (result, expect_params)
+        assert expect_params == result, msg
+
+
+class createMacro(Macro):
```

Comment:
I miss the execution using spock syntax string, e.g. "pt6_base 99 [1 2]" or "pt6_base 99 1 2". The same applies for execMacro macro.

Reply:
Solved at [commit 6923430](https://github.com/sardana-org/sardana/pull/781/commits/6923430d779d98e034b2f19930d42d0e6d6e5185)